youtube_dl/extractor/wdr.py

```diff
@@ -95,80 +82,87 @@ class WDRIE(InfoExtractor):
-            ]
+        if entries is None:  # Overview page
+            return self._overiew_page_extractor(page_url, page_id, webpage)
 
-            if entries:  # Playlist page
-                return self.playlist_result(entries, page_id)
+        jsonpage = self._download_webpage(entries.group(0), entries.group(0))
+        jsonvars = json.loads(jsonpage[38:-2])
 
-            # Overview page
-            entries = []
-            for page_num in itertools.count(2):
-                hrefs = re.findall(
-                    r'<li class="mediathekvideo"\s*>\s*<img[^>]*>\s*<a href="(/mediathek/video/[^"]+)"',
-                    webpage)
-                entries.extend(
-                    self.url_result(page_url + href, 'WDR')
-                    for href in hrefs)
-                next_url_m = re.search(
-                    r'<li class="nextToLast">\s*<a href="([^"]+)"', webpage)
-                if not next_url_m:
-                    break
-                next_url = page_url + next_url_m.group(1)
-                webpage = self._download_webpage(
-                    next_url, page_id,
-                    note='Downloading playlist page %d' % page_num)
-            return self.playlist_result(entries, page_id)
+        page_id = jsonvars['trackerData']['trackerClipId']
+        title = jsonvars['trackerData']['trackerClipTitle']
+        formats = []
+        for _id, video_field in jsonvars['mediaResource'].items():
+            if 'videoURL' in video_field:
+                video_url = video_field['videoURL']
+            elif 'audioURL' in video_field:
+                video_url = video_field['audioURL']
+            else:
+                break
+            is_live = video_field.get('flashvarsExt', {'isLive': '0'})
+            is_live = is_live.get('isLive', '0') == '1'
 
-        flashvars = compat_parse_qs(
-            self._html_search_regex(r'<param name="flashvars" value="([^"]+)"', webpage, 'flashvars'))
+            if video_url.endswith('.f4m'):
+                video_url += '?hdcore=3.2.0&plugin=aasp-3.2.0.77.18'
+                ext = 'flv'
+            elif video_url.endswith('.smil'):
+                fmt = self._extract_smil_formats(video_url, page_id)[0]
+                video_url = fmt['url']
+                sep = '&' if '?' in video_url else '?'
+                video_url += sep
+                video_url += 'hdcore=3.3.0&plugin=aasp-3.3.0.99.43'
+                ext = fmt['ext']
+            else:
+                ext = determine_ext(video_url)
 
-        page_id = flashvars['trackerClipId'][0]
-        video_url = flashvars['dslSrc'][0]
-        title = flashvars['trackerClipTitle'][0]
-        thumbnail = flashvars['startPicture'][0] if 'startPicture' in flashvars else None
-        is_live = flashvars.get('isLive', ['0'])[0] == '1'
+            formats.append({'url': video_url, 'ext': ext, 'format_id': _id})
```

Comment:
Extraction from JSON may result in duplicate download links like in your audio test URL.